tests/acceptance/characters/test_freerider_attacks.py

```diff
@@ -127,12 +127,12 @@ def test_pay_a_flunky_instead_of_the_arranged_ursula(blockchain_alice,
     # Enrico becomes
     enrico = Enrico(policy_encrypting_key=bupkiss_policy.public_key)
     plaintext = b"A crafty campaign"
-    message_kit, _signature = enrico.encrypt_message(plaintext)
+    message_kit = enrico.encrypt_message(plaintext)
 
-    with pytest.raises(RestMiddleware.PaymentRequired):
-        blockchain_bob.retrieve(message_kit,
+    with pytest.raises(Ursula.NotEnoughUrsulas):
```

Comment:
Based on the name of this test, how did the exception change?

Reply:
Filed #2789 - possibly for another PR? Note though that if we decide to save errors in `RetrievalResult`, this will change the protocol.